tests/puppeteer/pages/commonPage.js

```diff
@@ -57,4 +57,51 @@ module.exports = class CommonPage {
     await this.page.waitForSelector('body');
     return this.page;
   }
+
+  /**
+   * Wait for selector and click
+   * @param selector, element to check
+   * @return boolean, true if text exist, false if not
+   */
+  async waitForSelectorAndClick(selector, timeout = 10) {
+    await this.page.waitForSelector(selector, {visible: true, timeout});
+    await this.page.click(selector);
+  }
+
+  /**
+   * Check text value
+   * @param selector, element to check
+   * @param textToCheckWith, text to check with
+   * @param parameter, parameter to use
+   * @return boolean, true when the result of parameter(textToCheckWith) is true, false if not
```

Comment:
Same here This function does not return a boolean, it returns a Promise and throw an error if element does not exist or text is not correct